mne/tests/test_cov.py

```diff
@@ -435,20 +435,17 @@ def test_compute_covariance_auto_reg():
     events_merged = merge_events(events, event_ids, 1234)
     picks = pick_types(raw.info, meg='mag', eeg=False)
     epochs = Epochs(raw, events_merged, 1234, tmin=-0.2, tmax=0,
-                    picks=picks[:5], baseline=(-0.2, -0.1), proj=True,
+                    picks=picks[:7], baseline=(-0.2, -0.1), proj=True,
                     reject=reject, preload=True)
     epochs.crop(None, 0)[:10]
 
-    method_params = dict(factor_analysis=dict(iter_n_components=[30]),
-                         pca=dict(iter_n_components=[30]))
+    method_params = dict(factor_analysis=dict(iter_n_components=[3]),
+                         pca=dict(iter_n_components=[3]))
```

Comment:
using more components than number of picks does not make sense. That was the original issue.